basepage/base_page.py

```diff
@@ -241,6 +241,37 @@ def enter_text(self, locator, text, with_click=True, with_clear=False, with_ente
 
         actions.perform()
 
+    def erase_text(self, locator, click=True, clear=False, backspace=0, params=None):
+        """
+        Various ways to erase text from web element.
+
+        :param locator: locator tuple or WebElement instance
+        :param click: clicks the input field
+        :param clear: clears the input field
+        :param backspace: how many times to hit backspace
+        :param params: (optional) locator params
+        :return: None
+        """
+        element = locator
+        if not isinstance(element, WebElement):
+            element = self.get_visible_element(locator, params)
+
+        if click:
+            self.click(element)
+
+        actions = ActionChains(self.driver)
+
+        if clear:
+            element.clear()
+
+        if backspace:
+            times = 1
```

Comment:
I would also change this segment to simply do 'for _ in range(backspace):' 

Fewer lines and you let the interpreter optimize the variable and counter.